### ISSUE TITLE
Implement Recipes

### ISSUE THREAD

**Issue by fhnaumann** [MockBukkit/MockBukkit#1233]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Implement basic recipe support. For now this includes implementing `getCraftingRecipe` in `ServerMock` and changing `getRecipe` to include the default minecraft recipes.
Subsequent PR's could implement the `craft...` methods, but I think its best to separate these for now.

**Comment by Thorinwasher** [comment]
This seems like it would need some data mining to get all the minecraft recipes. Not doing this automatically would be a pain.

**Comment by 4everTheOne** [comment]
Breaking down the Minecraft recipes we have this type of recipes but each type of recipes can have sub-variants.

- blasting
- campfire_cooking
- crafting
- smelting
- smithing
- smoking
- stonecutting

In PR https://github.com/MockBukkit/MockBukkit/pull/1283 i tried to create the base code for this implementation using the crafting as example.

**Comment by fhnaumann** [comment]
Yeah I pulled all recipes from the game automatically. However, I gave up on the PR because I struggled to implement shapeless recipes properly.
